core/src/labels/labelMesh.cpp

```diff
@@ -66,30 +62,34 @@ void LabelMesh::loadQuadIndices() {
                  reinterpret_cast<GLbyte*>(indices.data()), GL_STATIC_DRAW);
 }
 
-void LabelMesh::compileVertexBuffer() {
-    size_t sumVertices = 0;
-    int stride = m_vertexLayout->getStride();
-    m_glVertexData = new GLbyte[stride * m_nVertices];
+void LabelMesh::compile(std::vector<std::unique_ptr<Label>>& _labels,
+                        std::vector<Label::Vertex>& _vertices) {
 
-    for (auto& vertices : m_vertices) {
-        size_t nVertices = vertices.size();
+    constexpr size_t maxVertices = 16384;
 
-        std::memcpy(m_glVertexData + (sumVertices * stride),
-                    reinterpret_cast<GLbyte*>(vertices.data()), nVertices * stride);
-        sumVertices += nVertices;
-    }
+    typedef std::vector<std::unique_ptr<Label>>::iterator iter_t;
+
+    m_labels.reserve(_labels.size());
```

Comment:
`insert` should take care of `reserve`. Or does it not when using a `move_iterator`?